multicluster/controllers/multicluster/core/remote_common_area.go

```diff
@@ -0,0 +1,345 @@
+				break
+			}
+		}
+	}
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancelFunc()
+	if localClusterMemberAnnounceExist {
+		if r.remoteCommonAreaManager.GetElectedLeaderClusterID() != common.InvalidClusterID {
+			localClusterMemberAnnounce.LeaderClusterID = string(r.remoteCommonAreaManager.GetElectedLeaderClusterID())
+		}
+
+		if localClusterMemberAnnounce.Annotations == nil {
+			localClusterMemberAnnounce.Annotations = make(map[string]string)
+		}
+		localClusterMemberAnnounce.Annotations["touch-ts"] = time.Now().String()
+		// do an update
+		if err := r.Update(ctx, &localClusterMemberAnnounce, &client.UpdateOptions{}); err != nil {
+			klog.ErrorS(err, "error updating MemberClusterAnnounce", "Cluster", r.GetClusterID())
+			return err
+		}
+	} else {
+		// Create happens first before a leader can be elected. When the creation is successful,
+		// it marks the connectivity status and then the leader election can happen.
+		// Therefore, the first create will not populate the leader ClusterID.
+		localClusterMemberAnnounce.ClusterID = string(r.remoteCommonAreaManager.GetLocalClusterID())
+		localClusterMemberAnnounce.Name = "member-announce-from-" + string(r.remoteCommonAreaManager.GetLocalClusterID())
+		localClusterMemberAnnounce.Namespace = r.Namespace
+		localClusterMemberAnnounce.ClusterSetID = string(r.ClusterSetID)
+		if err := r.Create(ctx, &localClusterMemberAnnounce, &client.CreateOptions{}); err != nil {
+			klog.ErrorS(err, "Error creating MemberClusterAnnounce", "Cluster", r.GetClusterID())
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *remoteCommonArea) updateRemoteCommonAreaStatus(connected bool, err error) {
+	defer r.mutex.Unlock()
+	r.mutex.Lock()
+	if r.connected == connected {
+		return
+	}
+
+	klog.InfoS("Updating RemoteCommonArea status", "Cluster", r.GetClusterID(), "Connected", connected)
+
+	// TODO: Tolerate transient failures so we dont oscillate between connected and disconnected.
+	r.connected = connected
+}
+
+/**
+ * ---------------------------
+ * commonArea Implementation
+ * ---------------------------
+ */
+
+func (r *remoteCommonArea) GetClusterID() common.ClusterID {
+	return r.ClusterID
+}
+
+func (r *remoteCommonArea) GetNamespace() string {
+	return r.Namespace
+}
+
+/**
+ * ---------------------------
+ * RemoteCommonArea Implementation
+ * ---------------------------
+ */
+
+// Start starts a background routine.
+// Once connected to the RemoteCommonArea, the Start method runs a timer
+// on a go routine to periodically write MemberClusterAnnounce into the
+// RemoteCommonArea's CommonArea and also maintain its connectivity status.
+func (r *remoteCommonArea) Start() (context.CancelFunc, error) {
+	stopCtx, stopFunc := context.WithCancel(context.Background())
+
+	// Start a Timer for every 5 seconds
+	// TODO: make the interval longer? the webhook API is called every 5s to RemoteCommonArea now
+	ticker := time.NewTicker(5 * time.Second)
+
+	go func() {
```

Comment:
I think `SendMemberAnnounce()` need to be called one time immediately before it goes into go routine, otherwise, when this `Start()` method is called, it actually will wait 5s to `SendMemberAnnounce()`

Reply:
Okay sure, changed.